proposed/dotnet-sdk-unified-config.md

```diff
@@ -0,0 +1,150 @@
+
+The choice of format is in some ways limited by engineering considerations. The easiest thing would be to choose a format that is already supported by the BCL, which would lead us to JSON, XML, or INI. JSON and XML formats have downsides, though: JSON doesn't support comments and has an anemic data model, and XML is historically verbose and difficult to parse in a consistent way.
+
+The recent wave of programming languages and programming-related tools have popularized the use of more human-readable formats like YAML, TOML, and INI. These formats are more human-readable and have more expressive power than JSON or XML. None of them have parsers in the BCL, though, which would mean that we would have to write our own parser or use a third-party library. Writing a parser is a potentially error-prone task, and using a third-party library would require onboarding the parser to source build as well as doing a thorough security audit. YAML specifically has such a lax parser that it might be a security risk to use it.
+
+My personal suggestion is that we should use [TOML][toml] and advocate for a parser for it in the BCL. TOML is in wide use in the [Python][python-why-toml] and Rust ecosystems. It strikes a good balance between human-readability and expressiveness. The [toml spec][toml-spec] includes a [comparison][toml-comparison] between TOML and other formats.
+
+### Tooling support
+
+#### CLI command
+CLI commands (dotnet config) a la `git config` will be provided to allow users to set and get configuration values. This will allow users to set configuration values without having to edit the file directly. We should allow reading of configuration values via a CLI command to provide a least-common-denominator access mechanism.
+
+`dotnet config get <key> [--local|--global]`
+
+For example, to get the 'realized' value of the `dotnet.telemetry` key for a given repo, you could run the following command in that repo:
+```shell
+> dotnet config get dotnet.telemetry
+true
+```
+and the following command would get the user-global value only
+```shell
+> dotnet config get dotnet.telemetry --global
+false
+```
+
+The [git config][git-config] command is a good source of inspiration.
+
+
+#### Programmatic access
+
+We (BCL) should provide a library to parse the file and integrate its data into `Microsoft.Extensions.Configuration` so that tools can easily consume the configuration data.
+
+#### IDE onboarding
+
+We must ensure that both the CLI and VS support the configuration file and the configuration system.
+
+#### Usage within the dotnet CLI and supporting tools
+Inside the `dotnet` CLI (and tools that ship within the CLI), we will initialize a `ConfigurationBuilder` with the `poohbear.config` file and merge it with the user's `poohbear.config` file. We will also seed that builder with environment variables. Tools that need the configuration within the `dotnet` CLI will read the appropriate values and apply them to their execution.
```

Comment:
If the tool / format starts to support env vars or other sources, consider adding a `dotnet config dump` or `dotnet config get --all` command or similar so that folks can debug setup issues.